src/App.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import List from './List';
+import Tabs from './components/Tabs';
+import firebase from "firebase/compat";
+import { useCollection } from "react-firebase-hooks/firestore";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+
+import './styles.css';
+import {filterType, modeType} from './Constants';
+
+const firebaseConfig = {
+  apiKey: "AIzaSyATi5uqr6dkf4iTvqfPHndGwtHXz0O6O-s",
+  authDomain: "lab-90286.firebaseapp.com",
+  databaseURL: "https://lab-90286-default-rtdb.firebaseio.com",
+  projectId: "lab-90286",
+  storageBucket: "lab-90286.appspot.com",
+  messagingSenderId: "782831139249",
+  appId: "1:782831139249:web:2d846e81abb155eeece590",
+  measurementId: "G-YTZ7W2BRD9"
+};
+
+firebase.initializeApp(firebaseConfig);
+const db = firebase.firestore();
+
+function App() {
+  const collection = db.collection('todo_list');
+  const [query, setQuery] = useState(collection);
+
+  const [todoListData, setTodoListData] = useState([]);
+  const [lastId, setLastId] = useState("");
+  const [filter, setFilterType] = useState(filterType.showAll);
+  const [mode, setMode] = useState(modeType.base);
+
+  const [value, error, loading] = useCollection(collection);
+  const [currList, setCurrList] = useState("");
+
+  useEffect(() => {
+    let data = [];
+    if (value) {
+      data = value.docs.map((doc) => {
+            return {...doc.data()}});
+      setTodoListData(data);
+      if (currList === "") {
+        setCurrList(data[0].list);
+      }
+    }
+  }, [currList, value]);
+
+  function plusClicked() {
+    // Adds an empty Todo
+    const newId = generateUniqueID();
+    let currDate = firebase.firestore.Timestamp.now();
+    collection.doc(newId).set({ id: newId, todo: "", completed: false, priority: 2, creationdate: currDate, list: currList});
+    setLastId(newId);
+    setMode(modeType.add);
+  }
+
+  function newList() {
+    // Creates a new list called "List<some combination of unique ID numbers>" with a "First Todo"
```

Comment:
Mixing two different kinds of things (lists and tasks) in the same collection is probably not the best approach.